File: HelloTuiLua/Resources/src/scene/welcome/dialog/msgBox.lua
require "extern"
require "tagMap/Tag_welcomeui"

MsgBox = class("MsgBox",function()
	return TuiBase:create()
end)

MsgBox.__index = MsgBox
local __instance

function MsgBox:create()
	local ret = MsgBox.new()
	TuiManager:sharedManager():parseScene(ret,"panel_msgbox",PATH_WELCOMEUI)
	ret:onLoadComplete()
	__instance = ret
	return ret
end

function MsgBox:getControl(tagPanel,tagControl)
	local ret = nil
	ret = self:getPanel(tagPanel):getChildByTag(tagControl)
	return ret
end

function MsgBox:getPanel(tagPanel)
	local ret = nil
	if tagPanel == Tag_welcomeui.PANEL_MSGBOX then
		ret = self:getChildByTag(tagPanel)
	end
	return ret
end

---------------logic----------------------------
local btnClose 		= nil
local labAtlasNum 	= nil
local progHp  		= nil
local sliderTest 	= nil

local function event_btn_close(p_sender)
	local window = tolua.cast(__instance:getParent(), "CWidgetWindow")
	window:setModalable(false)
	__instance:removeFromParentAndCleanup(true)
end

local function event_prog_hp(p_sender, n_value)
	labAtlasNum:setString(n_value)
	print("progTest :"..n_value)
end

local function event_slider_test(p_sender, n_value)
	print("sliderTest :"..n_value)
	progHp:setValue(n_value)
end

function MsgBox:onLoadComplete()
	--注册事件
	labAtlasNum = tolua.cast(self:getControl(Tag_welcomeui.PANEL_MSGBOX,Tag_welcomeui.LABATLAS_NUM),"CLabelAtlas")

	btnClose = tolua.cast(self:getControl(Tag_welcomeui.PANEL_MSGBOX,Tag_welcomeui.BTN_CLOSE),"CButton")
	btnClose:setOnClickScriptHandler(event_btn_close)

	progHp = tolua.cast(self:getControl(Tag_welcomeui.PANEL_MSGBOX,Tag_welcomeui.PROG_HP),"CProgressBar")
	progHp:setShowValueLabel(true)
	progHp:setOnValueChangedScriptHandler(event_prog_hp)

	sliderTest = tolua.cast(self:getControl(Tag_welcomeui.PANEL_MSGBOX,Tag_welcomeui.SLIDER_TEST),"CSlider")
	sliderTest:setOnValueChangedScriptHandler(event_slider_test)
end

File: HelloTuiCpp/Classes/scene/Main/dialog/MsgBox.cpp
﻿#include "MsgBox.h"
#include "../../tui/TuiManager.h"
#include "tagMap/tuiTag_main.h"

NS_MAIN_BEGIN

void MsgBox::onLoadScene()
{
	setAutoRemoveUnusedTexture(true);
	TuiManager::sharedManager()->parseScene(this, "panel_msgbox", PATH_MAIN);
	//注册事件
	CButton *pBtnClose = (CButton*)getControl(PANEL_MSGBOX,BTN_CLOSE);
	pBtnClose->setOnClickListener(this,ccw_click_selector(MsgBox::event_btn_close));
	
	CSlider *pSlider = (CSlider*)getControl(PANEL_MSGBOX,SLIDER_TEST);
	pSlider->setOnValueChangedListener(this,ccw_valuechanged_selector(MsgBox::event_slider_test));

	CProgressBar *pProg = (CProgressBar*)this->getControl(PANEL_MSGBOX,PROG_HP);
	pProg->setShowValueLabel(true);
}

void MsgBox::event_btn_close( CCObject *pSender )
{
	//解除塑形
	//CCSceneManager::sharedManager()->getRunningScene()->setModalable(false);

	CCSceneManager::sharedManager()->popUIScene(this);
}

void MsgBox::event_slider_test(CCObject *pSender, int value)
{
	char buf[128] = {0};
	sprintf(buf,"%d",value);
	CLabelAtlas *pLab = (CLabelAtlas*)this->getControl(PANEL_MSGBOX,LABATLAS_NUM);
	pLab->setString(buf);

	CProgressBar *pProg = (CProgressBar*)this->getControl(PANEL_MSGBOX,PROG_HP);
	pProg->setValue(value);

}

/************************************************************************/
//	GET/SET/IS
/************************************************************************/
CCNode *MsgBox::getPanel( int tagPanel )
{
	CCNode *pPanel = nullptr;
	switch (tagPanel)
	{
	case PANEL_MSGBOX:
		pPanel = this->getChildByTag(PANEL_MSGBOX);
		break;
	}
	return pPanel;
}


NS_MAIN_END